bin/webpack

```diff
@@ -7,12 +7,12 @@ require "yaml"
 ENV["RAILS_ENV"] ||= "development"
 RAILS_ENV = ENV["RAILS_ENV"]
 
-ENV["NODE_ENV"] ||= RAILS_ENV
+ENV["NODE_ENV"] ||= RAILS_ENV == "integration" ? "production" : RAILS_ENV
 NODE_ENV = ENV["NODE_ENV"]
 
 APP_PATH          = File.expand_path("../", __dir__)
 NODE_MODULES_PATH = File.join(APP_PATH, "node_modules")
-WEBPACK_CONFIG    = File.join(APP_PATH, "config/webpack/#{NODE_ENV}.js")
+WEBPACK_CONFIG    = File.join(APP_PATH, "config/webpack/#{RAILS_ENV}.js")
```

Comment:
I don't _**think**_ this will cause any issues with this change...

Reply:
well, webpack config uses `NODE_ENV` to load the right `webpacker.yml` section,
and to enable v2v specific development code.

it doesn't use `RAILS_ENV`

If we don't use `NODE_ENV=...` when invoking webpack during appliance build, we're probably fine :)

Reply:
> If we don't use `NODE_ENV=`... when invoking webpack during appliance build, we're probably fine :)

Yeah I just wasn't sure if there was a situation where developers were setting `NODE_ENV` specifically _**and**_ setting `RAILS_ENV`  to something that different.  This would break that workflow if running a bunch of tasks in succession.

You can of course work around it by triggering `bin/webpack`, or the tasks that run it, separately with the given environments, but if there isn't a use case where we would see an environment mismatch (besides the newly added `integration` environment), then I don't see an issue.

* * *

Also, for the record:  My assumption is that `node` enforces and only allows `test`, `development`, and `production` for `NODE_ENV`, so I assume that I have to stick with one of those, correct?

Reply:
> My assumption is that node enforces and only allows test, development, and production for NODE_ENV, so I assume that I have to stick with one of those, correct?

Not really, those are meaningless for node itself, and for webpack. (Just a common convention.)

So, this is just our code, `rake webpacker:compile` sets it to `Webpacker.env`, whatever that is, and then `config/webpack/configuration.js` reads it to load the right bit of the yaml,
and we pass it to our JS where only v2v checks it for `development`.